src/cascade/model/session.py

```diff
@@ -0,0 +1,552 @@
+        # Options in grey were rejected by Dismod-AT despite being in docs.
+        # https://bradbell.github.io/dismod_at/doc/option_table.htm
+        option = pd.DataFrame([
+            dict(option_name="parent_node_id", option_value=str(self.location_func(self.parent_location))),
+            dict(option_name="parent_node_name", option_value=nan),
+            dict(option_name="meas_std_effect", option_value="add_std_scale_all"),
+            dict(option_name="zero_sum_random", option_value=nan),
+            dict(option_name="data_extra_columns", option_value=nan),
+            dict(option_name="avgint_extra_columns", option_value=nan),
+            dict(option_name="warn_on_stderr", option_value="true"),
+            dict(option_name="ode_step_size", option_value="5.0"),
+            dict(option_name="age_avg_split", option_value=nan),
+            dict(option_name="random_seed", option_value="0"),
+            dict(option_name="rate_case", option_value="iota_pos_rho_zero"),
+            dict(option_name="derivative_test_fixed", option_value="none"),
+            dict(option_name="derivative_test_random", option_value="none"),
+            dict(option_name="max_num_iter_fixed", option_value="100"),
+            dict(option_name="max_num_iter_random", option_value="100"),
+            dict(option_name="print_level_fixed", option_value=5),
+            dict(option_name="print_level_random", option_value=5),
+            dict(option_name="accept_after_max_steps_fixed", option_value="5"),
+            dict(option_name="accept_after_max_steps_random", option_value="5"),
+            dict(option_name="tolerance_fixed", option_value="1e-8"),
+            dict(option_name="tolerance_random", option_value="1e-8"),
+            dict(option_name="quasi_fixed", option_value="false"),
+            dict(option_name="bound_frac_fixed", option_value="1e-2"),
+            dict(option_name="limited_memory_max_history_fixed", option_value="30"),
+            dict(option_name="bound_random", option_value=nan),
+        ], columns=["option_name", "option_value"])
+        self.dismod_file.option = option.assign(option_id=option.index)
+
+    def _create_node_table(self, locations):
+        columns = dict(
+            node_name=locations.name,
+            parent=locations.parent,
+        )
+        # This adds c_location_id, if it's there.
+        for add_column in [c for c in locations.columns if c.startswith("c_")]:
+            columns[add_column] = locations[add_column]
+        table = pd.DataFrame(columns)
+        table["node_id"] = table.index
+
+        def location_to_node_func(location_id):
+            if np.isnan(location_id):
+                return np.nan
+            return np.where(table.c_location_id == location_id)[0][0]
+
+        table["parent"] = table.parent.apply(location_to_node_func)
+        self.dismod_file.node = table
+        self.location_func = location_to_node_func
+
+
+class FitResult:
```

Comment:
I like this pattern of result objects.

Reply:
Yes! Comes from R, to my knowledge. It lets you get data conditionally, but make that transparent to the user. There could be a problem if you hold onto the object and change the dismodfile. That isn't handled here.